test/core/bad_client/tests/unknown_frame.cc

```diff
@@ -36,10 +35,21 @@ int main(int argc, char** argv) {
   grpc_init();
   grpc::testing::TestEnvironment env(argc, argv);
 
-  /* test adding prioritization data */
-  GRPC_RUN_BAD_CLIENT_TEST(verifier, nullptr,
-                           PFX_STR "\x00\x00\x00\x88\x00\x00\x00\x00\x01",
-                           GRPC_BAD_CLIENT_DISCONNECT);
+  /* test that all invalid/unknown frame types are handled */
+  for (int i = 10; i <= 255; i++) {
+    std::string unknown_frame_string;
+    unknown_frame_string.append("\x01\x01\x01", sizeof("\x00\x00\x00") - 1);
```

Comment:
Can we use just simple append(s) instead of append(s, n)?

Reply:
I just noticed that elements from debugging were left over. The issue with simply using append(s) is that the string is treated as a null-terminated C string which is not the case for us.

Reply:
Oh I see and it'd better to make a helper function to get only one string to improve readability and avoid mistakes. This could be done by using macro or template like

```
// Using macro
#define APPEND_BUFFER(buf) append(buf, sizeof(buf) - 1)
unknown_frame_string.APPEND_BUFFER("\x01\x01\x01");
```

```
// Using template
template<size_t N> void append_buffer(string, char (&)[N]) { ... }
append_buffer(unknown_frame_string, "\x01\x01\x01");
```

Reply:
done, used a macro.

A function would not work since that would convert the array to a char *, and sizeof on that would give the size of the pointer instead of the array.

Reply:
Good!